NDTensors/src/dense/dense.jl

```diff
@@ -89,8 +89,8 @@ function Base.real(::Type{Dense{ElT,DataT}}) where {DataT<:AbstractArray{ElT}} w
    default_Densetype( adapt(real(ElT), DataT))
 end
 
-function complex(::Type{Dense{ElT,DataT{ElT}}}) where {ElT}
-  return Dense{complex(ElT),DataT{complex(ElT)}}
+function complex(::Type{Dense{ElT,Vector{ElT}}}) where {ElT}
+  return Dense{complex(ElT),Vector{complex(ElT)}}
```

Comment:
Seems like this could be:
```julia
function complex(T::Type{<:Dense})
  return set_datatype(T, similartype(datatype(T), complex(eltype(T))))
end
```

Reply:
(Same for `real` above.)

Reply:
I've matched `real` and `complex`